flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2683,13 +2683,16 @@ class ScalarExprLowering {
   ExtValue genCopyIn(const ExtValue &actualArg,
                      const Fortran::lower::CallerInterface::PassedEntity &arg,
                      CopyOutPairs &copyOutPairs,
-                     llvm::Optional<mlir::Value> restrictCopyAtRuntime) {
+                     llvm::Optional<mlir::Value> restrictCopyAtRuntime,
+                     bool byValue) {
+    const bool doCopyOut = !byValue && arg.mayBeModifiedByCall();
+    llvm::StringRef tempName = byValue ? ".copy" : ".copyinout";
```

Comment:
Not for this PR:

We could/should consider having some functionality in the uniquer for temp names that come up. There are some cases where the same temporary name is used many times, which sort of defeats the purpose of having a name.